ddl/ddl.go

```diff
@@ -143,7 +143,8 @@ type DDL interface {
 		ctx sessionctx.Context,
 		schema model.CIStr,
 		info *model.TableInfo,
-		onExist OnExist) error
+		onExist OnExist,
+		stmt *ast.CreateTableStmt) error
```

Comment:
Why we need to add this parameter?
All the necessary information have been converted to TableInfo already....

Reply:
@tiancaiamao it is to be able to update the QueryString just before the DoDDLJob() call with the generated syntax.

I.e. the stmt parameter is needed to get access to the original syntactic sugar starting offset and length, so it can be replaced by the generated one.

If it is OK to change the QueryString without set it back, I can remove this change. How are the sessionctx.QueryString used throughout the query execution? Does it need to be reset to the original query, or should it simply be replaced with the non-syntactic sugar generated part?

Reply:
I reverted this, meaning smaller change, but the QueryString will permanently have the syntactic sugar replaced with the generated one for CREATE TABLE.